Core.Moodle.Interface/IMoodleClient.cs

```diff
@@ -5,12 +5,15 @@ public interface IMoodleClient
     /*
      * Get Course of student
      */
-    Task<MoodleCourse> GetCourses(string token);
+    IEnumerable<MoodleCourse> GetCourses(string token);
```

Comment:
```suggestion
    Task<IEnumerable<MoodleCourse>> GetCourses(string token);
```
To keep this function async `Task` is still needed. The other functions should also use Task so that they can be `async` which is more performant.